cocos/spine/skeleton.ts

```diff
@@ -1288,8 +1341,10 @@ export class Skeleton extends UIRenderer {
      * @method setEndListener
      * @param {function} listener
      */
-    public setEndListener (listener: TrackListener) {
+    public setEndListener (listener: any) {
```

Comment:
All listener APIs have been marked as any type, it's very unfriendly for public APIs, we should try to keep them, even if we fake a type. But I do think we can use wasm type imported from spine-core.d.ts